testsuite/run_test

```diff
@@ -0,0 +1,175 @@
+#!/bin/sh
+
+# from http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -eu
+#set -o pipefail
+#IFS=$' \n\t'
+#set -x # or -o xtrace
+#set -v # or -o verbose
+
+#
+# run_test script_name codes_success codes_failure
+#
+# script_name: name of a .sli / .py script absolute path or relative to build directory
+#
+# codes_success: variable that contains an explanation string for
+#                all exit codes that are to be regarded as a success
+# codes_skipped: variable that contains an explanation string for
+#                all exit codes that mean the test was skipped
+# codes_failure: variable that contains an explanation string for
+#                all exit codes that are to be regarded as a failure
+# Examples:
+#
+#   codes_success=' 0 Success'
+#   codes_skipped=\
+#   ' 200 Skipped,'\
+#   ' 201 Skipped (MPI required),'\
+#   ' 202 Skipped (build with-mpi=OFF required),'\
+#   ' 203 Skipped (Threading required),'\
+#   ' 204 Skipped (GSL required),'\
+#   ' 205 Skipped (MUSIC required),'
+#   codes_failure=\
+#   ' 1 Failed: missed assertion,'\
+#   ' 2 Failed: error in tested code block,'\
+#   ' 3 Failed: tested code block failed to fail,'\
+#   ' 126 Failed: error in test script,'
+#
+# Description:
+#
+#   The function runs the NEST binary with the SLI script script_name.
+#   The exit code is then transformed into a human readable string
+#   (if possible) using the global variables CODES_SUCCESS, CODES_SKIPPED, and
+#   CODES_FAILURE which contain a comma separated list of exit codes
+#   and strings describing the exit code.
+#
+#   If any of the variables CODES_SUCCESS, CODES_SKIPPED or CODES_FAILURE
+#   contain an entry for the exit code, the respective string is logged, 
+#   and the test is either counted as passed or failed.
+#
+#   If none of the variables CODES_SUCCESS, CODES_SKIPPED or CODES_FAILURE
+#   contain an entry != "" for the returned exit code, the pass is counted as
+#   failed, too, and unexpected exit code is logged).
+#
+#   Additionally this script needs TEST_PASSED, TEST_SKIPPED, TEST_FAILED, and
+#   TEST_TOTAL to be defined in the environment. Their values are treated as
+#   filenames where one line is added for the passed/skipped/failed test and
+#   one line in total.
+#
+# measure runtime of command
+#
+time_cmd()
+{
+    start=`date +%s%N`
```

Comment:
This will give the number of seconds with `N` appended on OSX and then arithmetic below works no more. `date +%s` gives just the seconds since the epoch, so by dropping the division below, one gets a coarse measure. Not sure how to get ns-resolution time on OSX.